src/main/src/test/java/org/geoserver/catalog/impl/CatalogImplTest.java

```diff
@@ -3514,6 +3514,165 @@ public void testFullTextSearchAddedKeyword() {
+    }
+
+    @Test
+    public void testExactTermSearchKeywords() {
+        ft.getKeywords().add(new Keyword("air_temp"));
+        ft.getKeywords().addAll(List.of(new Keyword("temperature"), new Keyword("Air")));
+        cv.getKeywords().add(new Keyword("dwpt_dprs"));
+        cv.getKeywords()
+                .addAll(List.of(new Keyword("temperature"), new Keyword("DewpointDepression")));
+
+        l.setResource(ft);
+        addLayer();
+        catalog.add(cs);
+        catalog.add(cv);
+        LayerInfo l2 = newLayer(cv, s);
+        catalog.add(l2);
+
+        Filter filter = Predicates.exactTermSearch("temperature");
+        assertEquals(newHashSet(l, l2), asSet(catalog.list(LayerInfo.class, filter)));
+        assertEquals(newHashSet(ft, cv), asSet(catalog.list(ResourceInfo.class, filter)));
+        assertEquals(newHashSet(ft), asSet(catalog.list(FeatureTypeInfo.class, filter)));
+        assertEquals(newHashSet(cv), asSet(catalog.list(CoverageInfo.class, filter)));
+
+        filter = Predicates.exactTermSearch("air");
+        assertEquals(newHashSet(l), asSet(catalog.list(LayerInfo.class, filter)));
+        assertEquals(newHashSet(ft), asSet(catalog.list(ResourceInfo.class, filter)));
+        assertEquals(newHashSet(ft), asSet(catalog.list(FeatureTypeInfo.class, filter)));
+        assertTrue(asSet(catalog.list(CoverageInfo.class, filter)).isEmpty());
+
+        filter = Predicates.exactTermSearch("dewpointdepression");
+        assertEquals(newHashSet(l2), asSet(catalog.list(LayerInfo.class, filter)));
+        assertEquals(newHashSet(cv), asSet(catalog.list(ResourceInfo.class, filter)));
+        assertTrue(asSet(catalog.list(FeatureTypeInfo.class, filter)).isEmpty());
+        assertEquals(newHashSet(cv), asSet(catalog.list(CoverageInfo.class, filter)));
+
+        filter = Predicates.exactTermSearch("pressure");
+        assertTrue(asSet(catalog.list(LayerInfo.class, filter)).isEmpty());
+        assertTrue(asSet(catalog.list(ResourceInfo.class, filter)).isEmpty());
+        assertTrue(asSet(catalog.list(FeatureTypeInfo.class, filter)).isEmpty());
+        assertTrue(asSet(catalog.list(CoverageInfo.class, filter)).isEmpty());
+    }
+
+    @Test
+    public void testExactTermSearchAddedKeyword() {
+        ft.getKeywords().add(new Keyword("air_temp"));
+        ft.getKeywords().add(new Keyword("temperatureAir"));
+
+        l.setResource(ft);
+        addLayer();
+
+        LayerInfo lproxy = catalog.getLayer(l.getId());
+        FeatureTypeInfo ftproxy = (FeatureTypeInfo) lproxy.getResource();
+
+        ftproxy.getKeywords().add(new Keyword("newKeyword"));
+        catalog.save(ftproxy);
+
+        Filter filter = Predicates.exactTermSearch("newKeyword");
+        assertEquals(newHashSet(ftproxy), asSet(catalog.list(FeatureTypeInfo.class, filter)));
+        assertEquals(newHashSet(lproxy), asSet(catalog.list(LayerInfo.class, filter)));
```

Comment:
I would add a test before the keyword being added, which should result in an empty result

Reply:
Added a test for a search miss on a different keyword. Thanks.